src/http/one/Tokenizer.cc

```diff
@@ -24,11 +25,11 @@ Http::One::Tokenizer::quotedString(SBuf &returnedToken, const bool http1p0)
 bool
 Http::One::Tokenizer::quotedStringOrToken(SBuf &returnedToken, const bool http1p0)
 {
-    checkpoint();
-
-    if (!skip('"'))
-        return prefix(returnedToken, CharacterSet::TCHAR);
-
+    if (!skip('"')) {
+        if (prefix(returnedToken, CharacterSet::TCHAR))
+            return true;
+        throw TexcHere("invalid token");
```

Comment:
Roughly speaking, Tokenizer methods that return bool should not throw to indicate problems. The caller is asking whether something is X. The correct reaction to finding something other than X is to return false, preferably without any side effects.

Reply:
> Roughly speaking, Tokenizer methods that return bool should not throw to indicate problems

How do you suggest then to distinguish these two situations (in both the qdText() returns false): 
1. Incorrect input (error)
2. More data is needed

We could return, e.g., an integer value instead of bool, or pass another variable by reference, indicating an error. Both approaches do not seem elegant, though.